app/views/cookies/show.html.erb

```diff
@@ -0,0 +1,94 @@
+<div class="govuk-grid-row">
+  <div class="govuk-grid-column-two-thirds">
+    <% if @consent_updated %>
+      <div class="govuk-notification-banner govuk-notification-banner--success" role="alert" aria-labelledby="govuk-notification-banner-title" data-module="govuk-notification-banner">
+        <div class="govuk-notification-banner__header">
+          <h2 class="govuk-notification-banner__title" id="govuk-notification-banner-title">
+            Success
+          </h2>
+        </div>
+        <div class="govuk-notification-banner__content">
+          <p class="govuk-notification-banner__heading">
+            You’ve set your cookie preferences.
+          </p>
+        </div>
+      </div>
+    <% end %>
+
+    <h1 class="govuk-heading-l">Cookies</h1>
+    <p>Cookies are small files saved on your phone, tablet or computer when you visit a website.</p>
+    <p>We use cookies to make the Early Career Framework (ECF) work and collect information about how you use our service.</p>
+
+    <h2 class="govuk-heading-m">Essential cookies</h2>
+    <p>Essential cookies keep your information secure while you use ECF. We do not need to ask permission to use them. </p>
+
+    <table class="govuk-table">
+      <caption class="govuk-visually-hidden">Essential cookies</caption>
+      <thead class="govuk-table__head">
+        <tr class="govuk-table__row">
+          <th class="govuk-table__header">Name</th>
+          <th class="govuk-table__header">Purpose</th>
+          <th class="govuk-table__header">Expires</th>
+        </tr>
+      </thead>
+      <tbody class="govuk-table__body">
+        <tr class="govuk-table__row">
+          <td class="govuk-table__cell">_govuk_rails_boilerplate_session</td>
+          <td class="govuk-table__cell" width="50%">Used to keep you signed in</td>
+          <td class="govuk-table__cell">2 weeks</td>
+        </tr>
+        <tr class="govuk-table__row">
+          <td class="govuk-table__cell">cookie_consent_1</td>
```

Comment:
any suggestions?

Reply:
Uff. 

I think this cookie is used for two different things. One, explicitly (in name and structure) is for 'has the user agreed to analytics cookies?'. The other is 'has the user interacted with us at all, therefore removing our need for the cookie banner?'

I am not convinced we want these to be the same cookie.

Maybe we should have a separate `cookie_banner_hidden` cookie, and another `analytics_consent` one?

Reply:
What's wrong with having a single cookie storing a boolean consent value? What would that give you extra?

Reply:
Are you suggesting that clicking hide should hide the banner forever?

Reply:
They store information about fundamentally different things. 

The question is: Why would you use a cookie that stores the user's consent to analytics to check whether or not you should show them the cookie banner?

The answer is, I guess "because that's the longest lived out of our cookies, and I don't bother having a separate cookie for hiding cookie banner". Which is not all that convincing to me.

Also, since this cookie is used for two different things, naming it in such a way that it is not lying about what it's used for is hard.

Reply:
When would we show them the cookie banner at a time other than to get their consent for analytics cookies?

Or in the future when other analytics cookies are added, when would we want to show the banner other than to get updated consent?

Reply:
I don't agree with this @kicferk1. We show the cookie banner when consent has not been stored. So we should check if consent has been stored to decide if we should show the banner.

Reply:
If we were to also decide to show the banner on tuesdays, or if the user clicks a box saying "always show me the annoying banner" then that would be different

Reply:
I imagine whenever we add more cookies we would want to re-show them the banner, even if the new cookies are not related to analytics at all.

I guess that kind of answers the question of "why is it called `consent_1`?". Cause whenever we add more cookies we will bump this to be `consent_2`, `consent_3`, and so on - so users see the banner again.

To me consent to analytics cookies and presenting the cookie banner are at best loosely related. If we were planning to add another type of behaviour that the user can agree to, and stored that in a cookie, then using a single cookie to show content banner and to store analytics cookie would wipe their previous choice on analytics. Which I guess doesn't matter much, because we won't store much in cookies. 

It's a happy coincidence that analytics cookie is set precisely when we want to hide the cookie banner - one that isn't always true. We could decide to hide that banner for 5 years, but only keep analytics for one year - if we wanted to - and change the phrasing of the banner based on whether it needs to inform about all cookies or just the analytics one. 

Like I said, I don't think it particularly matters, cause we might never get into more cookies. And I got the answer to why there is a `1` in the cookie name, as ugly as it is.

Reply:
But even in that case, if they've consented to analytics cookies, we only need to show them the banner if they need to give their consent again, and vice versa.

The only case where having them separate would make sense to me is if GA was replaced with something else in the future and we had to ask for consent again—but until they gave consent, carry on serving GA to them. Even with the maintenance cost of doing that aside, I don't think we'd ever want to do that.

Reply:
gov.uk does it: https://www.gov.uk/help/cookie-details

Reply:
It also has a separate cookie called `cookies_preferences_set`...

Reply:
We need to show the users the banner whenever we need their consent to some cookies. 

And I guess if we added another cookie, I don't know, 'keep me logged in for a month rather than 2 weeks' or something, then we would make the banner re-appear.

To do that, we would bump `consent_1` to `consent_2`. And by doing that, we make it a lot harder to know what they answered to the analytics question - cause it lives on `consent_1`, but some other things live on `consent_2`, and it leads to chaos and destruction each time we add more things to cookies. 

Will we? No, probably not. Would it be a good idea to separate the two anyway? Yes, cause you can give things more meaningful names.

Reply:
No. This is a versioned consent cookie. When they have completed version one of consent, we never need show them the banner again unless something changes. In which case, everything is stored in the `consent_2`. `consent_1` dies and it's information is transfered

Reply:
- I arrive at the site
- I consent to analytics
- consent_1 = yes
- banner is not shown in future visits
- We decide to have a non-essential settings cookie
- banner reappears
- I consent to both
- delete(consent_1)
- consent_2 = { analytics: yes, settings: yes }

Reply:
It would be nice to show them 'last time you agreed to analytics, and nothing around that changed' - if we let them opt in one but not the other.

To do that you need some code reading your `consent_1` cookie.

And how long will that code live in your app? A year :P 

I don't really understand why you would not want to have a separate cookie for remembering that the user set their cookie preferences, and another to store what those were. The gov.uk page you linked to does that - they have `cookies_policy` for storing the details of consent, and `cookies_preferences_set` - I imagine precisely to make things more readable and obvious.

Reply:
Why not refer to the previous cookie if you want to show them that they agreed last time?

Reply:
> Why not refer to the previous cookie if you want to show them that they agreed last time?

A user logs in in May. They agree to analytics cookie, we store it in `consent_1`. We add another thing to consent to, and a new cookie in June, call it `consent_2`. 

We don't know when the user logs in again - could be next week, could be in a few months. If we want to show them what they agreed to last time, we need to have some code that's aware of what our cookie was called last time the user agreed to it. 

And we would need to have that code for roughly as long as the expiry time of `consent_1` cookie. 

Doing what govuk does, and having `cookies_policy` for storing cookies preferences and `cookies_preferences_set` for the cookie banner being shown lets us decouple them. We could change the `cookies_preferences_set` to a new name, or store a date that the cookie banner was hidden, or whatever - and we wouldn't need to worry about how it affects the existing analytics settings, or any other settings - cause they are in a different cookie. 

And again, I don't see what the risk or price of having two separate cookies for this stuff is. I guess slightly more code, but that should come with increase in readability.

Anyway, this feels like a massive exercise in bike-shedding. This solution will work for our needs almost certainly, and if it doesn't and we want to change it, then the worst that happens is slight frustration from the users seeing cookie banner too many times.